jans-cedarling/cedarling/src/jwt/test/utils.rs

```diff
@@ -0,0 +1,248 @@
+/*
+ * This software is available under the Apache-2.0 license.
+ * See https://www.apache.org/licenses/LICENSE-2.0.txt for full text.
+ *
+ * Copyright (c) 2024, Gluu, Inc.
+ */
+
+use std::time::{SystemTime, UNIX_EPOCH};
+
+use serde::{Deserialize, Serialize};
+
+use jsonwebkey::{self as jwk, Algorithm};
+use jsonwebtoken::{self as jwt};
+
+/// Represents the claims contained within a JWT.
+///
+/// This is a user-defined struct for holding token claims.
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+pub struct Claims {
+    iss: String,
+    sub: String,
+    aud: String,
+    exp: u64,
+    iat: u64,
+}
+
+/// Generates a JSON Web Token (JWT) with specified algorithm and expiration status.
+///
+/// # Arguments
+///
+/// * `algorithm` - The signing algorithm to use for the JWT -- only ES256 is working reliably for now
+/// * `expired` - A boolean indicating whether the token should be generated as expired.
+///
+/// # Returns
+///
+/// A tuple containing:
+/// - A `String` representing the serialized JWT.
+/// - A `String` representing the public JWK serialized as a String.
+/// - A `Claims` struct representing the claims of the token.
+pub fn generate_token(algorithm: jwt::Algorithm, expired: bool) -> (String, String, Claims) {
+    let jwk_alg = match algorithm {
+        jwt::Algorithm::HS256 => jwk::Algorithm::HS256,
+        jwt::Algorithm::ES256 => jwk::Algorithm::ES256,
+        jwt::Algorithm::RS256 => jwk::Algorithm::RS256,
+        _ => panic!("not implemented"),
+    };
+
+    let (enc_key, dec_key) = generate_key_pair(jwk_alg);
+
+    let claims = generate_claims(expired);
+    let header = jwt::Header {
+        typ: Some("JWT".to_string()),
+        alg: algorithm,
+        cty: None,
+        jku: None,
+        jwk: None,
+        kid: Some(enc_key.0.clone()),
+        x5u: None,
+        x5c: None,
+        x5t: None,
+        x5t_s256: None,
+    };
+
+    let token = jwt::encode(&header, &claims, &enc_key.1).expect("should generate token");
+    (token, dec_key, claims)
+}
+
+/// Generates claims for a JWT based on whether it should be expired.
+///
+/// # Arguments
+///
+/// * `expired` - A boolean indicating if the claims should be generated for an expired token.
+///
+/// # Returns
+///
+/// A `Claims` struct containing the issuer, subject, audience, issued at time, and expiration time.
+fn generate_claims(expired: bool) -> Claims {
+    let iat: u64;
+    let exp: u64;
+    if expired {
+        iat = 1695198692;
```

Comment:
I added some comments now. let me know if this is satisfactory:

```rust
  let iat: u64;
  let exp: u64;
  if expired {
      iat = 1695198692; // timestamp for issued at (iat)
      exp = 1695195092; // timestamp for expiration (exp)
  } else {
      let now = SystemTime::now();
      // set issued at (iat) at the current time
      iat = now.duration_since(UNIX_EPOCH).unwrap().as_secs();
      exp = iat + 3600; // set expiration 1 hour from now
  }
```

diff: [utils.rs](https://github.com/JanssenProject/jans/pull/9726/files#diff-17b91e99995d6737784ed4980ba0762686940b04159da5781c6c4ea89e750fcd)

Reply:
when I see number 1695198692 I can`t understand what data it is. It would be nice to have a description of unix time number. To understand that it is past.